sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -265,3 +270,61 @@ export function settleMessageOperation(
     throw translateServiceBusError(err);
   });
 }
+
+/** @internal */
+export interface RetryForeverArgs<T> {
+  retryConfig: RetryConfig<T>;
+  onError: (err: Error) => void;
+  logger: ReturnType<typeof createServiceBusLogger>;
+  logPrefix: string;
+}
+
+/**
+ * Retry infinitely until success, reporting in between retry<> attempts.
+ *
+ * This function will only stop retrying if:
+ * - args.retryConfig.operation resolves successfully
+ * - args.retryConfig.operation rejects with an `AbortError`
+ *
+ * @internal
+ */
+export async function retryForever<T>(
```

Comment:
Nit and not a blocker for this PR

`retryForever` is only used by the streaming receiver in non session case. So far other methods in this file "receiverCommon" are for cases that are indeed common between sessions and non sessions. So, perhaps not the right place for the code. Moving it can also re-enforce the understanding of one of the main differences in sessions and non sessions receiver which is the forever loop